pkg/sql/logictest/testdata/logic_test/interval

```diff
@@ -650,3 +650,96 @@ WHERE tablename = 'intervalstyle_in_index'
 AND indexname = 'idx'
 ----
 idx  CREATE INDEX idx ON test.public.intervalstyle_in_index USING btree (b ASC) WHERE (b > 'P3Y'::INTERVAL)
+
+# date_trunc
+statement ok
+SET intervalstyle = 'postgres'
+
+statement error pgcode 22023 interval units "invalid" not recognized
+SELECT
+  date_trunc('invalid', interval '1 month')
+
+query TTT
+SELECT
```

Comment:
optional, but if you wanted to make this table driven-y for non-errors, you could make it:

```
query TTT
SELECT i, truncate_str, date_trunc(truncate_str, i) FROM (VALUES
   ('decade', interval '1001 year 1 month 1 day 1 microsecond'),
   ('millennium', '1001 year 1 month 1 day 1 microsecond')
   -- etc
) AS t(truncate_str, i)
```